uniffi_bindgen/src/bindings/python/mod.rs

```diff
@@ -33,13 +33,33 @@ impl crate::BindingGenerator for PythonBindingGenerator {
         components: &mut Vec<Component<Self::Config>>,
     ) -> Result<()> {
         for c in &mut *components {
+            c.config
+                .module_name
+                .get_or_insert_with(|| c.ci.namespace().to_string());
             c.config.cdylib_name.get_or_insert_with(|| {
                 settings
                     .cdylib
                     .clone()
                     .unwrap_or_else(|| format!("uniffi_{}", c.ci.namespace()))
             });
         }
+        // We need to update module names
+        let packages = HashMap::<String, String>::from_iter(
+            components
+                .iter()
+                .map(|c| (c.ci.crate_name().to_string(), c.config.module_name())),
+        );
+        for c in components {
+            for (ext_crate, ext_package) in &packages {
+                if ext_crate != c.ci.crate_name()
+                    && !c.config.external_packages.contains_key(ext_crate)
+                {
+                    c.config
+                        .external_packages
```

Comment:
We don't have good tests for this, but I think this will break [this external type](https://github.com/mozilla/uniffi-rs/blob/c5da61baf98d421f2da521b1ab085f576695178d/uniffi_bindgen/src/bindings/python/gen_python/mod.rs#L139-L148) support when the module name is overridden.

For my interest, what's your use-case for using a different module name than the namespace, and are you using multiple crates?

Reply:
> For my interest, what's your use-case for using a different module name than the namespace, and are you using multiple crates?

No just a single crate, but the issue that I was running into is that the name of my ffi crate can't be the same as the dependency that I am creating the bindings for, which I ran into when starting to use proc-macros, as uniffi is confused about multiple dylibs of the same name.

Reply:
Can you just change the namespace name? It doesn't have to be the crate name.